File: cpp/LeetCode/No1.cpp
#include <iostream>
#include <vector>
#include <map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> mem;
        for(int i=0;i<nums.size();i++){
            int temp = target - nums[i];
            if(mem.find(temp) != mem.end()){
                return vector<int>({mem[temp], i});
            }
            mem[nums[i]] = i;
        }
        return vector<int>({-1});
    }
};

int main(){
    vector<int> nums = {3,2,4};
    int target = 6;

    Solution solution;
    vector<int> res = solution.twoSum(nums, target);

    for(int num : res){
        printf("%d, ", num);
    }
    printf("\r\n");

    return 0;
}

File: python/LeetCode Quiz/No1.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mem = dict()
        for i, val in enumerate(nums):
            temp = target - val
            if temp in mem:
                return [mem[temp], i]
            mem[val] = i
        return [-1]

def main():
    nums = [3,2,4]
    target = 6
    print(Solution.twoSum(Solution(), nums, target))

if __name__=='__main__':
    main()
